classes/controller/FrontController.php

```diff
@@ -477,12 +477,56 @@ public function init()
             $this->displayRestrictedCountryPage();
         }
 
+        if (Tools::isSubmit('live_edit') && !$this->checkLiveEditAccess()) {
+            Tools::redirect('index.php?controller=404');
+        }
+
         $this->iso = $iso;
         $this->context->cart = $cart;
         $this->context->currency = $currency;
     }
 
     /**
+     * Checks if the user can use Live Edit feature
+     *
+     * @return bool
+     */
+    public function checkLiveEditAccess()
+    {
+        if (!Tools::isSubmit('live_edit') || !Tools::getValue('ad') || !Tools::getValue('liveToken')) {
+            return false;
+        }
+
+        if (Tools::getValue('liveToken') != Tools::getAdminToken('AdminModulesPositions'.(int)Tab::getIdFromClassName('AdminModulesPositions').(int)Tools::getValue('id_employee'))) {
+            return false;
+        }
+
+        return is_dir(_PS_CORE_DIR_.DIRECTORY_SEPARATOR.Tools::getValue('ad'));
```

Comment:
Please consider adding some security about `Tools::getValue('ad')`. You should sanitize it + test `is_dir()` on a realpath, not a simple concatenated string.